tools/cargo_bazel/src/annotation.rs

```diff
@@ -181,10 +182,29 @@ impl LockfileAnnotation {
 
         // Handle any git repositories
         if let Some(git_ref) = source.git_reference() {
+            let mut manifest_dir = pkg.manifest_path.clone();
+            manifest_dir.pop();
+            let mut ancestors = manifest_dir.ancestors();
+            let mut git_root = None;
+            while let Some(ancestor) = ancestors.next() {
+                if ancestor.join(".git").is_dir() {
+                    git_root = Some(ancestor);
+                    break;
+                }
+            }
+            let git_root = git_root.unwrap_or_else(|| panic!("Expected git repository to have a .git directory for package {} with manifest {}", pkg.name, pkg.manifest_path));
```

Comment:
This function already returns a `Result`, maybe return some `anyhow::Context` instead of panicing?

Reply:
Good call - done